lblod/ember-rdfa-editor
# Fix issues with pressing delete-key before inline ember-nodes in firefox and chromium

### Overview
This PR fixes issues with pressing the delete-key before inline ember-nodes (such as links) in both firefox and chromium.
It does so by adding custom delete-keydown handlers to both the `firefoxCursorFix` and `chromeHacksPlugin` plugins.

##### connected issues and PRs:
None

### How to test/reproduce
Without this fix:

- Open the dummy app in firefox or chromium
- Type some characters, and then insert an empty link: `test<link>|`
- Set the cursor at the start:  `|test<link>`
- Delete all the characters before the link with the `delete` key
- In case of using firefox: the cursor appears after the link (instead of before it): `<link>|`
- In case of using chromium: a `br` element is added before the link node: `|<br><link>`

With this fix:

- Open the dummy app in firefox or chromium
- Type some characters, and then insert an empty link: `test<link>|`
- Set the cursor at the start:  `|test<link>`
- Delete all the characters before the link with the `delete` key
- The cursor appears before the link in both firefox and chromium (without a `br` element added): `|<link>`

### Challenges/uncertainties
You'll notice that the `delete` handler logic is the same for both the chromium and firefox plugins. I did not merge these plugins yet, but it could be useful to merge/restructure these plugins in a way.

### Checks PR readiness
